### DISCUSSION TITLE
Flask v11.4.0 simple keyring issues

### DISCUSSION THREAD

**Discussion by leopashov** [MetaMask/snaps#1905] [General]
I cloned metamask version 11.4.0 (https://github.com/MetaMask/metamask-extension) and built it from source following the readme.
Ran it using 'yarn start:flask', and installed to edge following the guide at https://github.com/MetaMask/metamask-extension/blob/develop/docs/add-to-chrome.md. I also uninstalled my previous version of metamask flask.

I then cloned and ran the snap-simple-keyring example (https://github.com/MetaMask/snap-simple-keyring) using 'yarn start'.
Upon loading localhost:8000 the console looks like this: 
![image](https://github.com/MetaMask/snaps/assets/58823129/78145032-c6e4-41c0-86ea-84c71d62c737)
And I get this in the 'background.html' console:
![image](https://github.com/MetaMask/snaps/assets/58823129/6a5fa9b2-141f-4a17-93e8-72a6a5a5dacb)
The background.html console shows: 
![image](https://github.com/MetaMask/snaps/assets/58823129/f28b0e3d-5d5a-41df-bb90-4ce9ed2703c6)
when approving to install the keyring snap,
and:
![image](https://github.com/MetaMask/snaps/assets/58823129/e0e9d860-acef-4ba8-a90c-4d87632793f6)
When clicking 'ok' after successful snap installation.
Clicking 'create account' on the front end gives this:
![image](https://github.com/MetaMask/snaps/assets/58823129/c2c07a28-e286-4cf4-9131-6dafdb821c45)
Do you have any suggestions for how to fix/ workaround this?
I am using WSL2 and Microsoft Edge Version 118.0.2088.76 (Official build) (64-bit), and would be very happy to provide any additional details (or better formatted logs!)

Thank you!

**Comment by leopashov** [comment]
for additional context, the 'network' tab looks like this upon loading: 
![WhatsApp Image 2023-11-01 at 09 23 45](https://github.com/MetaMask/snaps/assets/58823129/9419b740-9836-45ae-82af-d274ccbb386d)
where the 'page-data.json' looks like this:
![WhatsApp Image 2023-11-01 at 09 24 47](https://github.com/MetaMask/snaps/assets/58823129/5a1a4e8d-8f9e-4fc1-82be-44eecd15c85f)
Meanwhile the successful calls look like:
![image](https://github.com/MetaMask/snaps/assets/58823129/f768b514-5229-40f9-81c0-c1d0483ef6c2)
and:
![image](https://github.com/MetaMask/snaps/assets/58823129/73191b8b-3a39-46c7-837f-cae0fabc392d)

It also seems the error message relating to the localhost not being permitted originates from a getDefaultExternalMiddleware call which should be somewhere in the '@metamask/utils' library, however I cannot find where exactly. Hopefully some of this may be of help!

**Reply by Montoya** [reply]
Hi, if you are trying to install the Snap from localhost, that is not allowed on stable MetaMask Extension. You can never install Snaps from localhost on MetaMask stable, only Flask. Instead, you can install the Snap from here: https://metamask.github.io/snap-simple-keyring/latest/

**Reply by leopashov** [reply]
@Montoya Thanks for getting back to me - ok running from your link worked! Does that mean I'm using metamask stable? My metamask version from within the extension is given as '11.4.0-flask.0' - is this not a Flask installation? 
Really hoping to be able to build on snap-simple-keyring in localhost and feel I'm a bit closer but still a little confused.
Thank you very much!

**Reply by leopashov** [reply]
@Montoya Just added localhost:8000 to my manifest and worked! Thank you very much :)

**Reply by Montoya** [reply]
We call orange MetaMask "MetaMask stable" and purple MetaMask is "MetaMask Flask" but technically they are both stable :)

**Comment by leopashov** [comment]
So far so good, thank you
